src/moveit_visual_tools.cpp

```diff
@@ -438,7 +438,8 @@ bool MoveItVisualTools::publishAnimatedGrasps(const std::vector<moveit_msgs::Gra
       break;
 
     publishAnimatedGrasp(possible_grasps[i], ee_jmg, animate_speed);
-
+    if (batch_publishing_enabled_)
+      trigger();
```

Comment:
interesting. it seems to me this trigger actually belongs inside ``publishAnimatedGrasp()`` because it has its own for-loop that publishes the motion of the end effector. thoughts?

Reply:
done